src/main/java/org/openrewrite/staticanalysis/EqualsAvoidsNull.java

```diff
@@ -74,83 +80,95 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
                         new UsesMethod<>(EQUALS_IGNORE_CASE),
                         new UsesMethod<>(CONTENT_EQUALS)),
                 new JavaVisitor<ExecutionContext>() {
+
                     @Override
                     public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
-                        J.MethodInvocation m = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
+                        return visitMethodInvocation((J.MethodInvocation) super.visitMethodInvocation(method, ctx));
+                    }
 
-                        if (!isStringComparisonMethod(m) || !hasCompatibleArgument(m) || m.getSelect() instanceof J.Literal) {
+                    private J visitMethodInvocation(J.MethodInvocation m) {
```

Comment:
SOC - as here is the action happening; we get the data we **actually need** and extract the DTO stuff to outer world we do not care and know about.